src/services/codefixes/importFixes.ts

```diff
@@ -1400,11 +1404,21 @@ function promoteFromTypeOnly(
     switch (aliasDeclaration.kind) {
         case SyntaxKind.ImportSpecifier:
             if (aliasDeclaration.isTypeOnly) {
-                const sortKind = OrganizeImports.detectImportSpecifierSorting(aliasDeclaration.parent.elements, preferences);
-                if (aliasDeclaration.parent.elements.length > 1 && sortKind) {
+                if (aliasDeclaration.parent.elements.length > 1) {
                     const newSpecifier = factory.updateImportSpecifier(aliasDeclaration, /*isTypeOnly*/ false, aliasDeclaration.propertyName, aliasDeclaration.name);
-                    const comparer = OrganizeImports.getOrganizeImportsComparer(preferences, sortKind === SortKind.CaseInsensitive);
-                    const insertionIndex = OrganizeImports.getImportSpecifierInsertionIndex(aliasDeclaration.parent.elements, newSpecifier, comparer, preferences);
+                    const { comparersToTest, typeOrdersToTest } = OrganizeImports.getDetectionLists(preferences);
+                    let specifierComparer = OrganizeImports.getOrganizeImportsSpecifierComparer(preferences, comparersToTest[0]);
+
+                    if (typeof preferences.organizeImportsIgnoreCase !== "boolean" || !preferences.organizeImportsTypeOrder) {
+                        const importDecl: ImportDeclaration[] = [aliasDeclaration.parent.parent.parent];
+                        const namedImportSort = OrganizeImports.detectNamedImportOrganizationBySort(importDecl, comparersToTest, typeOrdersToTest);
+                        if (namedImportSort) {
+                            const { namedImportComparer, typeOrder } = namedImportSort;
+                            specifierComparer = OrganizeImports.getOrganizeImportsSpecifierComparer({ organizeImportsTypeOrder: typeOrder }, namedImportComparer);
+                        }
+                    }
+
+                    const insertionIndex = OrganizeImports.getImportSpecifierInsertionIndex(aliasDeclaration.parent.elements, newSpecifier, specifierComparer);

```

Comment:
Would it make sense to wrap some of this up into a higher-level helper function in organizeImports.ts? It feels like the ideal case is for auto-imports to just say “I want to insert this specifier into this import; just tell me where it should go”